js/src/components/google-combo-account-card/connect-mc/merchant-center-select.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { getSetting } from '@woocommerce/settings'; // eslint-disable-line import/no-unresolved
+
+/**
+ * Internal dependencies
+ */
+import MerchantCenterSelectControl from '.~/components/merchant-center-select-control';
+import useExistingGoogleMCAccounts from '.~/hooks/useExistingGoogleMCAccounts';
+import useGoogleMCAccount from '.~/hooks/useGoogleMCAccount';
+import AppSelectControl from '.~/components/app-select-control';
+
+/**
+ * Renders the connected Merchant Center details, leveraging existing functionality
+ * and styles from AppSelectControl.
+ *
+ * Ideally, we would use MerchantCenterSelectControl only for this purpose. However, during testing,
+ * we found that when a URL is reclaimed for Merchant Center (MC), the Google API does not
+ * return the newly reclaimed account in the list provided by the useExistingGoogleMCAccounts hook,
+ * even though the data in the store is invalidated.
+ *
+ * @param {Object} props
+ * @param {boolean} props.isConnected Whether the Merchant Center account is connected.
+ */
+const MerchantCenterSelect = ( { isConnected, ...rest } ) => {
+	const { data: existingAccounts } = useExistingGoogleMCAccounts();
+	const { googleMCAccount } = useGoogleMCAccount();
+	const domain = new URL( getSetting( 'homeUrl' ) ).host;
+
+	const accountIdExists = existingAccounts?.some(
+		( existingAccount ) => existingAccount.id === googleMCAccount.id
+	);
+
+	if ( ! accountIdExists && isConnected ) {
```

Comment:
This is a clever workaround. I assume this is for a newly created account that has not yet shown up in the API response for existing accounts? If so, adding that context to the inline doc would be helpful.

Reply:
@joemcgill I added it at the start of the component where we explain why we have this intermediary component:
https://github.com/woocommerce/google-listings-and-ads/pull/2639/files#diff-38d86396eccec4e8e7e5b37e8de9cf25e28fb0a2536261feb312adfe9c6689aeR15-R24